docs/designs/ps_design.md

```diff
@@ -1,377 +1,146 @@
+In PS mechanism, machines are designated as three roles, one master, multiple workers and a PS with one or more PS instances. In the training process, workers pull parameters from the PS and push gradients to the PS. The PS should update model parameters using gradients received.
 
-The master will create a PS with *N* PS nodes, where *N* is specified by the user. Each model variable and embedding vector has a corresponding PS node. Thus, every PS node has a subset of model variables and embedding tables.
+It is noticeable that most designs use multiple PS instances to form a *service* in each job. These instances can collaboratively maintain large models with model sharding. Sometimes, even when the model is not very big, users still want multiple PS instances to benefit performance, because a single PS instance might become the bottleneck of communication and computation.
 
-The master will monitor PS nodes status similar to what it does for workers. In case a PS node fails, the master will try to relaunch it. Since PS nodes have higher priority than workers, if there are still some running workers, the relaunch will succeed by using either idle or preempted Kubernetes resources. If the relaunch fails, there are no workers left. The Elastic job has to wait for resources for the PS node and workers.
+In literatures, we see works about parameter server designs that handle large models and improve performance. However, ElasticDL needs an additional property -- fault-tolerance. The rest of the document will focus on how to achieve the three goals simultaneously:
 
-Each worker has a local copy of the model variables. After the master relaunches a PS node, the PS node can recover model variables from workers. For embedding vectors, PS must create replicas to support fault tolerance. For each PS node *PS(i)*, it will store *M* replicas in the following *M* PS nodes from *PS(i+1 % N)* to *PS(i+M % N)*. The relaunched PS node can recover embedding vectors from one of its replicas. If there are more than *M* continuously-indexed PS nodes failing, at least one PS node fails with all of its replicas. The ElasticDL job has to recover from a recent checkpoint.
+1. model sharding
+2. high performance
+3. fault-tolerance
 
-## PS
+## Model Sharding
+There are two kinds of parameters in the training process, embedding parameters and non-embedding parameters.
 
-### Compoments
+Embedding parameters consist of multiple embedding tables. Each embedding table correponds to one embedding layer in the user-defined model structure. An embedding table is a data structure that maps a discrete value, named embedding id *id*, to a vector, named embedding vector *vector*. Since embedding parameters might be up to terabytes, we can distribute them by placing different embedding vectors on different parameter server instance. 
```

Comment:
`placing different embedding vectors on different parameter server instance.` gives the impression that every embedding vector has a different PS instance. 
May change to `distribute embedding vectors into multiple PS instances.`

Reply:
Done